src/components/Review/DecisionArea.tsx

```diff
@@ -9,129 +9,137 @@ import { SummaryViewWrapperProps } from '../../formElementPlugins/types'
 import useUpdateReviewResponse from '../../utils/hooks/useUpdateReviewResponse'
 
 interface DecisionAreaProps {
-  reviewResponse: ReviewResponse
-  toggle: boolean
   summaryViewProps: SummaryViewWrapperProps
+  reviewResponse: ReviewResponse
+  isConsolidation: boolean
+}
+
+const optionsMap = {
+  consolidation: [
+    {
+      label: strings.LABEL_CONSOLIDATION_AGREE,
+      decision: ReviewResponseDecision.Agree,
+    },
+    {
+      label: strings.LABEL_CONSOLIDATION_DISAGREE,
+      decision: ReviewResponseDecision.Disagree,
+    },
+  ],
+  review: [
+    {
+      label: strings.LABEL_REVIEW_APPROVE,
+      decision: ReviewResponseDecision.Approve,
+    },
+    {
+      label: strings.LABEL_REVIEW_RESSUBMIT,
+      decision: ReviewResponseDecision.Decline,
+    },
+  ],
 }
 
 const DecisionArea: React.FC<DecisionAreaProps> = ({
-  toggle,
-  reviewResponse,
   summaryViewProps,
+  reviewResponse: initialReviewResponse,
+  isConsolidation,
 }) => {
-  const { updateQuery } = useRouter()
-  const [isOpen, setIsOpen] = useState(false)
-  const [previousToggle, setPreviousToggle] = useState(false)
-  const [review, setReview] = useState(reviewResponse)
-  const updateResponse = useUpdateReviewResponse(reviewResponse.id)
-
-  useEffect(() => {
-    if (toggle != previousToggle) {
-      setReview(reviewResponse)
-      setPreviousToggle(toggle)
-      setIsOpen(true)
-    }
-  }, [toggle])
+  const {
+    query: { openResponse },
+    updateQuery,
+  } = useRouter()
+  const [reviewResponse, setReviewResponse] = useState(initialReviewResponse)
+  const updateResponse = useUpdateReviewResponse()
 
   const submit = async () => {
-    // TODO do we need to handle update error ?
-    await updateResponse(review)
     updateQuery({ openResponse: null })
+    // TODO do we need to handle update error ?
+    await updateResponse(reviewResponse)
   }
 
+  const isValidComment =
+    (!!reviewResponse.comment && reviewResponse.comment?.trim() !== '') ||
```

Comment:
(not required change): just realised we doing a lot of `!`, when generating this one and when this one is used. Maybe `isInvalidComment` is easier ? (if no comment && Disagree or Decline)

Reply:
97e507a